api/read/read.go

```diff
@@ -0,0 +1,31 @@
+// Package read implements request object readers.
+package read
+
+import (
+	"encoding/json"
+	"io"
+
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smallstep/certificates/errs"
+)
+
+// JSON reads JSON from the request body and stores it in the value
+// pointed by v.
+func JSON(r io.Reader, v interface{}) error {
+	if err := json.NewDecoder(r).Decode(v); err != nil {
+		return errs.BadRequestErr(err, "error decoding json")
+	}
+	return nil
+}
```

Comment:
Are you planning to add the validator later?

Reply:
Right - so, the plan is this: this is the first PR in a series of PRs which, in t he end, will also include the validator.

For now, I'm trying to implement these changes incrementally so what's happening may be reviewed easier.

Reply:
Final result should also render helpers like `errs.BadRequestErr` obsolete, not quite sure how I'll get there right now - but the rest should be straightforward.